Modules/arraymodule.c

```diff
@@ -685,14 +735,17 @@ array_richcompare(PyObject *v, PyObject *w, int op)
 
     if (k) {
         /* No more items to compare -- compare sizes */
+        equal_except_size: ;
```

Comment:
It confused me that this can be jumped into even when sizes are equal. I would suggest renaming `equal_except_size` to something else. Perhaps `compare_sizes`?